spec/Appendix A -- Persisted Documents.md

```diff
@@ -0,0 +1,256 @@
+colon symbols (`:`). The meaning of a custom document identifier is
+implementation specific.
+
+Note: A 32 character hexidecimal _custom document identifier_ is likely to be an
+MD5 hash of the GraphQL document, as traditionally used by Relay.
+
+## Persisting a Document
+
+A client utilising persisted documents MUST generate a _document identifier_ for
+each GraphQL Document it wishes to issue to the server, and SHOULD ensure that
+the server can retrieve this GraphQL Document given the document identifier.
+
+Note: The method through which the client and server achieve this is
+implementation specific. Typically persisted documents are stored into some kind
+of trusted shared key-value store at client build time (either directly, or
+indirectly via an authenticated request to the server) such that the server may
+retrieve them given the identifier at request time. It is generally good
+practice for the client to issue both the GraphQL Document and the document
+identifier to the server; the server should then regenerate the document
+identifier from the GraphQL Document independently, and check that the
+identifiers match before storing the Document. An alternative approach has the
+client issue the GraphQL Document to the server, and the server returns an
+arbitrary _custom document identifier_ that the client should incorporate into
+its bundle.
+
+When using persisted documents as an operation allowlist, the client MUST
+persist the documents it uses ahead of time in a secure manner (preventing
+untrusted third parties from adding their own persisted document) such that the
+server will be able to retrieve the identified document within a _persisted
+document request_ and know that it is trusted.
+
+## Persisted Document Request
+
+A server MAY accept a _persisted document request_ via `GET` or `POST`.
+
+### Persisted Document Request Parameters
+
+:: A _persisted document request_ is an HTTP request that encodes the following
+parameters in one of the manners described in this specification:
+
+- {documentId} - (_Required_, string): The string identifier for the Document.
+- {operationName} - (_Optional_, string): The name of the Operation in the
+  identified Document to execute.
+- {variables} - (_Optional_, map): Values for any Variables defined by the
+  Operation.
+- {extensions} - (_Optional_, map): This entry is reserved for implementors to
+  extend the protocol however they see fit.
+
+### GET
+
+For a _persisted document request_ using HTTP GET, parameters SHOULD be provided
+in the query component of the request URL, encoded in the
+`application/x-www-form-urlencoded` format as specified by the
+[WhatWG URLSearchParams class](https://url.spec.whatwg.org/#interface-urlsearchparams).
+
+The {documentId} parameter must be a string _document identifier_.
+
+The {operationName} parameter, if present, must be a string.
+
+Each of the {variables} and {extensions} parameters, if used, MUST be encoded as
```

Comment:
Do we have to define expectations in case this exceeds maximum URL size?

Reply:
If we do; we should do it in the main spec: https://graphql.github.io/graphql-over-http/draft/#sec-GET

The Appendix tries not to redundantly repeat statements from the main spec if it can avoid it.

Reply:
Which reminds me; I heard that some people are using headers to specify variables when using GraphQL-over-GET... Apparently that works around the length limit :raised_eyebrow:

Reply:
Yes, a lot of folks use headers and then the server adds them to the `Vary` response-headers so browsers/... can know that it is part of the cache-key